tachyon/math/base/simd_int_unittest.cc

```diff
@@ -0,0 +1,81 @@
+#include "tachyon/math/base/simd_int.h"
+
+#include "gtest/gtest.h"
+
+namespace tachyon::math {
+
+template <typename SimdInt>
+class SimdIntTest : public testing::Test {};
+
+using SimdIntTypes = testing::Types<SimdInt128>;
+TYPED_TEST_SUITE(SimdIntTest, SimdIntTypes);
+
+TYPED_TEST(SimdIntTest, Zero) {
+  using SimdInt = TypeParam;
+
+  EXPECT_TRUE(SimdInt().IsZero());
+  EXPECT_TRUE(SimdInt::Zero().IsZero());
+  EXPECT_FALSE(SimdInt::One().IsZero());
+}
+
+TYPED_TEST(SimdIntTest, One) {
+  using SimdInt = TypeParam;
+
+  EXPECT_FALSE(SimdInt().IsOne());
+  EXPECT_FALSE(SimdInt::Zero().IsOne());
+  EXPECT_TRUE(SimdInt::One().IsOne());
+}
+
+template <typename SimdInt, typename T>
+void TestBroadcast() {
+  using BigInt = typename SimdInt::value_type;
+
+  T v = base::Uniform(base::Range<T>());
+  BigInt expected;
+  for (size_t i = 0; i < BigInt::kByteNums / sizeof(T); ++i) {
+    reinterpret_cast<T*>(&expected)[i] = v;
+  }
+  BigInt actual = SimdInt::Broadcast(v).value();
+  EXPECT_EQ(actual, expected);
+}
+
+TYPED_TEST(SimdIntTest, Broadcast) {
+  using SimdInt = TypeParam;
+
+  TestBroadcast<SimdInt, uint8_t>();
+  TestBroadcast<SimdInt, uint16_t>();
+  TestBroadcast<SimdInt, uint32_t>();
+  TestBroadcast<SimdInt, uint64_t>();
+}
+
+TYPED_TEST(SimdIntTest, EqualityOperations) {
+  using SimdInt = TypeParam;
+
+  SimdInt a = SimdInt::Random();
+  SimdInt b = SimdInt::Random();
+  EXPECT_EQ(a, a);
+  EXPECT_NE(a, b);
```

Comment:
Isn't it theoretically possible that `a` and `b` are the same, although the probability would be very low?

Reply:
What about this instead?
```
  SimdInt a = SimdInt::Random();
  SimdInt b = SimdInt::Random();
  while (b != a) {
    b = SimdInt::Random();
  }
  EXPECT_EQ(a, a);
  EXPECT_NE(a, b);

Reply:
I'll add one on purpose to handle the case.